src/secrets_manager/secrets_manager.py

```diff
@@ -0,0 +1,272 @@
+
+    return out_str
+
+
+def generate_resource_prefix_from_deployment_tier(  # pylint: disable=invalid-name
+    deployment_tier: str,
+) -> str:
+    """Append the appropriate prefix to a resource name."""
+    if deployment_tier == "production":
+        return ""
+    now = datetime.datetime.utcnow()
+    hostname = remove_invalid_resource_name_charaters(socket.gethostname()).lower()
+    max_hostname_length = 5
+    if len(hostname) > max_hostname_length:
+        hostname = hostname[:max_hostname_length]
+    resource_prefix = f"zztest_{hostname}_{now.strftime('%y%m%d%H%M%S')}_"
+    return resource_prefix
+
+
+def get_deployment_tier_prefix(deployment_tier: str) -> str:
+
+    if deployment_tier in ("test", "testing"):
+        return "test_"
+    if deployment_tier in ("prod", "production"):
+        return "prod_"
+    raise UnrecognizedVaultDeploymentTierError(deployment_tier)
+
+
+def _prepend_specific_deployment_tier(stub: str, deployment_tier: str) -> str:
+    return "%s%s" % (get_deployment_tier_prefix(deployment_tier), stub)
+
+
+class Vault:
+    """Manages access to secrets through various sources.
+
+    First searches any internal secrets specifically set in the vault
+    after instantiation. Then searches the environmental variables, then
+    specified JSON-formatted files (if supplied during init).
+
+    If a boto3 AWS session has been supplied, and there is a secret configured called 'aws_parameter_store_prefix', then that prefix will be checked (after other checks) to see if the desired secret exists there.
+    """
+
+    def __init__(
+        self,
+        deployment_tier: str = "test",
+        search_environment_first: bool = True,
+        files_to_search: Optional[List[str]] = None,
+    ):
+        if files_to_search is None:
+            files_to_search = list()
+        self._files_to_search = files_to_search
+        self._search_environment_first = search_environment_first
+        self._validate_and_assign_deployment_tier(deployment_tier)
+        self._internal_secrets: Dict[str, Union[str, float, int]] = dict()
+
+        self._boto_session: Optional[str] = None  # an AWS boto3 session
+
+    def set_boto_session(self, session: str) -> None:
+        """Set a boto3 AWS session.
+
+        This boto3 connection will be used to obtain values from the AWS parameter store.
+
+        There is no type hint because this package does not actually import boto3 (to lighten dependencies).  The session can just be injected by code using this package that has imported boto3.
+        """
+        self._boto_session = session
+
+    def get_boto_session(self) -> object:
+        return self._boto_session
+
+    def get_deployment_tier(self) -> str:
+        return self._deployment_tier
+
+    def _validate_and_assign_deployment_tier(self, deployment_tier: str) -> None:
+        self._deployment_tier = deployment_tier
+        if deployment_tier in ["test", "testing", "prod", "production"]:
```

Comment:
if the value is `testing` or `production`, would this method be the right place to just convert it to `test` / `prod`?